sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -186,6 +186,18 @@ static bool _on_link(const ipv6_addr_t *dst, unsigned *iface)
     return ipv6_addr_is_link_local(dst);
 }
 
+static gnrc_netif_t *_acquire_new_iface(unsigned iface)
```

Comment:
Why not

```suggestion
static gnrc_netif_t *_acquire_new_iface(gnrc_netif_t *netif)
```

Looks like that when you are calling this, you already have a pointer to `netif`.

Reply:
Nope, as you can see, I am getting the `netif` pointer from the numerical `iface` below first. Sure, I can do that also before calling `_acquire_new_iface()`, but my thought was, if I do it in any case, why not do it in this function?

Reply:
In line 225 you already have the `netif` pointer

Reply:
To clarify: This function is used in the case when the provided `netif` is not the one the next hop is associated with. So, the `netif` I "already have", is a different one potentially, than the one that is returned by this function. That's why I need to release and then acquire `netif` again.

Reply:
> In line 225 you already have the `netif` pointer

Look at what the comment says: "release pre-assumed netif". This `netif` pointer is a different one from the one we fetch in the line below. The `if` condition in l272 should make clear what I mean.